src/types/ComponentType.ts

```diff
@@ -60,9 +73,11 @@ export class ComponentType extends InheritableType {
         }
     }
 
+    private hasStartedAddingBuiltInInterfaces = false;
```

Comment:
needed to add this incase there were some circular references -- there would be stack overflows